tests/bootstrap.php

```diff
@@ -96,21 +100,12 @@
 ]);
 
 // Ensure default test connection is defined
-if (!getenv('db_class')) {
-    putenv('db_class=Cake\Database\Driver\Sqlite');
-    putenv('db_dsn=sqlite::memory:');
+if (!getenv('db_dsn')) {
+    putenv('db_dsn=sqlite:///:memory:');
 }
 
-ConnectionManager::setConfig('test', [
-    'className' => 'Cake\Database\Connection',
-    'driver' => getenv('db_class'),
-    'dsn' => getenv('db_dsn'),
-    'database' => getenv('db_database'),
-    'username' => getenv('db_login'),
-    'password' => getenv('db_password'),
-    'timezone' => 'UTC',
-    'quoteIdentifiers' => getenv('quoteIdentifiers'),
-]);
+ConnectionManager::setConfig('test', ['url' => getenv('db_dsn')]);
+ConnectionManager::setConfig('test_custom_i18n_datasource', ['url' => getenv('db_dsn')]);
```

Comment:
There was no `test_custom_i18n_datasource` before.

Reply:
In the end of this file replace `Carbon/Chronos` lines with single `Chronos::setTestNow(Chronos::now());`.

Remove PHPUnit aliases.

Reply:
My issue is, that the tests expects data which is not defined in the fixtures. Like this:
```php
        $result = $Aco->node('controllers/Admin/Posts')->toArray();
        $this->assertEquals($result[0]['alias'], 'Posts');
        $result = $Aco->find('children', ['for' => $result[0]['id']])->toArray();
        $this->assertEquals(count($result), 3);
```
What am i missing here?

Reply:
I don't use this plugin so I can't help much on plugin functionality itself.

Code above these lines might be broken (`$this->Task->acoUpdate();` maybe).

I can help on generic 3 to 4 upgrade. There are [many breaking](https://book.cakephp.org/4/en/appendices/4-0-migration-guide.html) changes in 4.0, you might need to fix them first. Use step by step with debugger if necessary.